### PR TITLE
Hipp 772 send delete application emails

### PR DESCRIPTION
**Pull request by llmikeyj**

### CODE REVIEW

File: app/uk/gov/hmrc/apihubapplications/connectors/EmailConnectorImpl.scala [hunk 1]
```diff
@@ -43,6 +43,26 @@ class EmailConnectorImpl @Inject()(
     templateId
   }
 
+  private val applicationDeletedToCreatorTemplateId = {
+    val templateId = servicesConfig.getConfString("email.deleteApplicationEmailToCreatorTemplateId", "")
```
Comment (edgarjimenez) [inline app/uk/gov/hmrc/apihubapplications/connectors/EmailConnectorImpl.scala:47]: All of these `templateId` initialisations from config look quite similar. Consider extracting them to a common method

File: app/uk/gov/hmrc/apihubapplications/connectors/EmailConnectorImpl.scala [hunk 2]
```diff
@@ -77,8 +97,66 @@ class EmailConnectorImpl @Inject()(
...
+
+    val request = SendEmailRequest(
+      Seq(toCreator),
+      applicationDeletedToCreatorTemplateId,
+      Map(
+        "applicationname" -> application.name
+      )
+    )
+
+    httpClient.post(url)
```
Comment (edgarjimenez) [inline app/uk/gov/hmrc/apihubapplications/connectors/EmailConnectorImpl.scala:114]: This `post` could probably be extracted to a common method as well

File: test/uk/gov/hmrc/apihubapplications/services/ApplicationsServiceSpec.scala [hunk 3]
```diff
@@ -487,6 +490,54 @@ class ApplicationsServiceSpec
...
+        _ =>
+          val captor = ArgCaptor[Application]
+          verify(repository).delete(captor.capture)
+          verify(emailConnector).sendApplicationDeletedEmailToTeam(ArgumentMatchers.eq(application))(any())
+          verify(emailConnector).sendApplicationDeletedEmailToCreator(ArgumentMatchers.eq(application))(any())
+          succeed
+      }
+    }
+
+    "must tolerate email failure and return success" in {
```
Comment (edgarjimenez) [inline test/uk/gov/hmrc/apihubapplications/services/ApplicationsServiceSpec.scala:519]: We should also add a test to ensure emails are not sent if the delete application operation fails

File: app/uk/gov/hmrc/apihubapplications/connectors/EmailConnectorImpl.scala (hunk 1 above)
Comment (llmikeyj) [reply app/uk/gov/hmrc/apihubapplications/connectors/EmailConnectorImpl.scala:47]: Okay

File: app/uk/gov/hmrc/apihubapplications/connectors/EmailConnectorImpl.scala (hunk 2 above)
Comment (llmikeyj) [reply app/uk/gov/hmrc/apihubapplications/connectors/EmailConnectorImpl.scala:114]: Okay

File: test/uk/gov/hmrc/apihubapplications/services/ApplicationsServiceSpec.scala (hunk 3 above)
Comment (llmikeyj) [reply test/uk/gov/hmrc/apihubapplications/services/ApplicationsServiceSpec.scala:519]: Okay

**Review by edgarjimenez — DISMISSED** [review]

**Review by PaulCDurham — APPROVED** [review]
There's one new exception that's not used. Up to you if you want to remove it now or not. Maybe just do that in the next ticket?

**Pull request merged**